packages/react/src/components/TimePicker/TimePicker.tsx

```diff
@@ -42,44 +47,113 @@ export const TimePicker: React.FunctionComponent<ITimePickerProps> = ({
   timeRange,
   strings = getDefaultStrings(useHour12, showSeconds),
   defaultValue,
+  value,
+  dateAnchor,
   onChange,
   onFormatDate,
   onValidateUserInput,
+  onGetErrorMessage,
+  placeholder = strings.defaultTimePickerPlaceholder,
   ...rest
 }: ITimePickerProps) => {
-  const [userText, setUserText] = React.useState<string>('');
+  const [comboBoxText, setComboBoxText] = React.useState<string>('');
+  const [selectedKey, setSelectedKey] = React.useState<string | number | undefined>();
   const [errorMessage, setErrorMessage] = React.useState<string>('');
 
+  const fallbackDateAnchor = useConst(new Date());
+
+  const [dateStartAnchor, setDateStartAnchor] = React.useState<Date>(dateAnchor || defaultValue || new Date());
+  const [dateEndAnchor, setDateEndAnchor] = React.useState<Date>(dateAnchor || defaultValue || new Date());
+
+  const [selectedTime, setSelectedTime] = useControllableValue(value, defaultValue, (_ev, newValue) =>
+    onChange?.(undefined, newValue),
+  );
+
   const optionsCount = getDropdownOptionsCount(increments, timeRange);
 
-  const initialValue = useConst(defaultValue || new Date());
-  const baseDate: Date = React.useMemo(() => generateBaseDate(increments, timeRange, initialValue), [
-    increments,
-    timeRange,
-    initialValue,
+  const internalDateAnchor = React.useMemo(() => dateAnchor || value || defaultValue || fallbackDateAnchor, [
```

Comment:
Just wondering, is it redundant to have an `internalDateAnchor` on top of the regular `dateAnchor` prop?

Reply:
For this to be a non-breaking change we cannot require `dateAnchor` and need to fall back to the other values when it is not provided to ensure consistent behavior.